.github/workflows/test.yml

```
@@ -83,8 +83,16 @@ jobs:
       - name: Documentation (Scala 2.12 only)
         if: matrix.scala == '2.12.13'
         run: sbt ++${{ matrix.scala }} docs/mdoc
-      - name: Test
+      - name: Test with treadle (PR only)
+        if: github.event_name == 'pull_request'
         run: sbt ++${{ matrix.scala }} test noPluginTests/test
+        env:
+          TEST_BACKEND: treadle
+      - name: Test (Push only)
+        if: github.event_name == 'push'
+        run: sbt ++${{ matrix.scala }} test noPluginTests/test
+        env:
+          TEST_BACKEND: verilator
```

Instead of having two tests guarded by an `if`, can we instead just set the environment variable in bash? I don't like that this could result in different tests being run for PR vs. push if someone updates one but not the other by accident. Also this suppresses tests for all events but PR and push and that's not right, you also want it on tagging and there might be other events too.

Something like:
```suggestion
      - name: Test
        run:
          - if [ $GITHUB_EVENT_NAME = "pull_request" ]; then export TEST_BACKEND="treadle"; else export TEST_BACKEND="verilator"; fi
          - sbt ++${{ matrix.scala }} test noPluginTests/test
```

or a shorter line:

```suggestion
      - name: Test
        run:
          - [ $GITHUB_EVENT_NAME = "pull_request" ] && TB="treadle" || TB="verilator"
          - export TEST_BAKCNED=$TB
          - sbt ++${{ matrix.scala }} test noPluginTests/test
```